### PR TITLE
BHBC-1219: Initial publish survey + scrape occurrences code.

### PR DESCRIPTION
**Pull request by NickPhura**
- Includes some misc naming updates.

# Overview

## Links to jira tickets

https://quartech.atlassian.net/browse/BHBC-1219

## This PR contains the following changes

Publish survey button
 - Endpoint fetches the latest occurrence submission
 - Scrapes out the occurrence data (assumes its a darwin core zip file currently)
 - Inserts data into occurrence table
 - Updates survey table to set publish_date

Misc small updates to some function names, etc.

## This PR contains the following types of changes

- [x] New feature (change which adds functionality)
- [ ] Enhancement (improvements to existing functionality)
- [ ] Bug fix (change which fixes an issue)
- [ ] Misc cleanup / Refactoring / Documentation
- [ ] Version change

## Checklist

A list of items that are good to consider when making any changes.

_Note: this list is not exhaustive, and not all items are always applicable._

### General

- [ ] I have performed a self-review of my own code

### Code

- [ ] New files/classes/functions have appropriately descriptive names and comment blocks to describe their use/behaviour
- [ ] I have avoided duplicating code when possible, moving re-usable pieces into functions
- [ ] I have avoided hard-coding values where possible and moved any re-usable constants to a constants file
- [ ] My code is as flat as possible (avoids deeply nested if/else blocks, promise chains, etc)
- [ ] My code changes account for null/undefined values and handle errors appropriately
- [ ] My code uses types/interfaces to help describe values/parameters/etc, help ensure type safety, and improve readability

### Style

- [ ] My code follows the established style conventions
- [ ] My code uses native material-ui components/icons/conventions when possible

### Documentation

- [ ] I have commented my code sufficiently, such that an unfamiliar developer could understand my code
- [ ] I have added/updated README's and related documentation, as needed

### Tests

- [ ] I have added/updated unit tests for any code I've added/updated
- [ ] I have added/updated the Postman requests/tests to account for any API endpoints I've added/updated

### Linting/Formatting

- [ ] I have run the linter and fixed any issues, as needed  
       _See the `lint` commands in package.json_
- [ ] I have run the formatter and fixed any issues, as needed  
       _See the `format` commands in package.json_

### SonarCloud

- [ ] I have addressed all SonarCloud Bugs, Vulnerabilities, Security Hotspots, and Code Smells

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Screenshots

Please add any relevant UI screenshots, if applicable.

### CODE REVIEW

File: api/src/errors/CustomError.ts [hunk 1]
```diff
@@ -41,3 +41,20 @@ export class HTTP500 extends CustomError {
...
   }
 }
+
+/**
+ * Ensures that the error is a `CustomError`.
+ * If `error` is a `CustomError`, then change nothing and return it.
+ * If `error` is not a `CustomError`, wrap it into an `HTTP500` error.
+ *
+ * @param {Error} error
+ * @param {strign} [message]
```
Comment (sdevalapurkar) [inline api/src/errors/CustomError.ts]:
```suggestion
 * @param {string} [message]
```

File: api/src/paths/project/{projectId}/survey/{surveyId}/publish.test.ts [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+      send: () => {
+        // do nothing
+      }
+    };
+  }
+} as any;
+
+const sampleNext = {} as any;
+
+describe('project/{projectId}/survey/{surveyId}/publish', () => {
```
Comment (sdevalapurkar) [inline api/src/paths/project/{projectId}/survey/{surveyId}/publish.test.ts]:
```suggestion
describe('publishSurveyAndOccurrences', () => {
```

File: api/src/queries/survey/survey-occurrence-queries.ts [hunk 3]
```diff
@@ -43,7 +43,107 @@ export const postSurveyOccurrenceSubmissionSQL = (
...
+    message: 'sql',
+    'sqlStatement.text': sqlStatement.text,
+    'sqlStatement.values': sqlStatement.values
+  });
+
+  return sqlStatement;
+};
+
+/**
+ * SQL query to get all occurrence submissions for a survey.
```
Comment (sdevalapurkar) [inline api/src/queries/survey/survey-occurrence-queries.ts]:
```suggestion
 * SQL query to get latest occurrence submission for a survey.
```

File: app/src/features/projects/view/ProjectPage.test.tsx [hunk 4]
```diff
@@ -137,37 +143,6 @@ describe('ProjectPage', () => {
     });
   });
 
-  it('sees delete project button as disabled when accessing a published project as a project administrator', async () => {
```
Comment (sdevalapurkar) [inline app/src/features/projects/view/ProjectPage.test.tsx:140]: question: why was this removed?

**Review by sdevalapurkar — COMMENTED** [review]
awesome code
really great test additions too!

just a few comments/questions but looks good

File: app/src/features/projects/view/ProjectPage.test.tsx (hunk 4 above)
Comment (NickPhura) [reply app/src/features/projects/view/ProjectPage.test.tsx:140]: It was a duplicate test, from what I could tell.  Same test name and same assertions, and same setup

Comment (sdevalapurkar) [reply app/src/features/projects/view/ProjectPage.test.tsx:140]: Ah I see, ok cool

**Review by sdevalapurkar — APPROVED** [review]
